pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -213,6 +216,58 @@ func (nc *NodeController) RegisterNodes(nodes *api.NodeList, retryCount int, ret
 	}
 }
 
+// reconcileExternalServices updates balancers for external services, so that they will match the nodes given.
+func (nc *NodeController) reconcileExternalServices(nodes *api.NodeList) {
+	balancer, ok := nc.cloud.TCPLoadBalancer()
+	if !ok {
+		glog.Error("The cloud provider does not support external TCP load balancers.")
+		return
+	}
+
+	zones, ok := nc.cloud.Zones()
+	if !ok {
+		glog.Error("The cloud provider does not support zone enumeration.")
+		return
+	}
+	zone, err := zones.GetZone()
+	if err != nil {
+		glog.Errorf("Error while getting zone: %v", err)
+		return
+	}
+
+	hosts := []string{}
+	for _, node := range nodes.Items {
+		hosts = append(hosts, node.Name)
+	}
+
+	services, err := nc.kubeClient.Services(api.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		glog.Errorf("Error while listing services: %v", err)
+		return
+	}
+	for _, service := range services.Items {
+		if service.Spec.CreateExternalLoadBalancer {
+			nonTCPPort := false
+			for i := range service.Spec.Ports {
+				if service.Spec.Ports[i].Protocol != api.ProtocolTCP {
+					nonTCPPort = true
+					break
+				}
+			}
+			if nonTCPPort {
+				// TODO: Support UDP here.
+				glog.Errorf("External load balancers for non TCP services are not currently supported: %v.", service)
+				continue
+			}
+			name := cloudprovider.GetLoadBalancerName(nc.clusterName, service.Namespace, service.Name)
+			err := balancer.UpdateTCPLoadBalancer(name, zone.Region, hosts)
+			if err != nil {
+				glog.Errorf("External error while updating TCP load balancer: %v.", err)
```

Comment:
I think it is better to wait a while (e.g.: 5 seconds) before the retry. If we assume that the error is transient, it is better not to spam cloud provider with our requests, but wait and give some time, so the error condition is over.

I've added an error log with info about the re-transmission.